### PR TITLE
Introduce a new Function that checks for equality for 3D vecs

### PR DESCRIPTION
**Pull request by Davknapp**
Will make testing vectors easier for upcoming tests

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: src/t8_vec.h [hunk 1]
```diff
@@ -186,6 +186,20 @@ t8_vec_diff (const double vec_x[3], const double vec_y[3], double diff[3])
...
+ * \param[in] vec_y 
+ * \param[in] precision 
+ * \return true, if the vectors are equal up to \a precision 
+ */
+static inline int
+t8_vec_eq (const double vec_x[3], const double vec_y[3], const double precision)
+{
+  return t8_vec_dist (vec_x, vec_y) < precision;
+}
+
```
Comment (holke) [inline src/t8_vec.h:202]: Why are you not checking |x[i] - y[i]| < eps?
Seems like less computational effort...

Comment (Davknapp) [reply src/t8_vec.h:202]: Changed it to an elementwise check.

**Review by lukasdreyer — COMMENTED** [review]
I like the addition, only have a more general naming convention comment

File: test/t8_gtest_custom_assertion.hxx [hunk 2]
```diff
@@ -65,4 +66,36 @@ element_equality (const char *ts_expr, const char *elem_1_expr, const char *elem
...
+                                        << " is not equal to \n"
+                                        << vec_2[0] << " " << vec_2[1] << " " << vec_2[2] << " " << vec_2_expr << " \n"
+                                        << "Precision given by " << precision_expr << " " << precision;
+#else
+    return testing::AssertionFailure () << vec_1_expr << " is not equal to " << vec_2_expr;
+#endif
+  }
+}
+
+#define EXPECT_VEC_EQ(vec_1, vec_2, precision) EXPECT_PRED_FORMAT3 (vec_equality, (vec_1), (vec_2), (precision))
```
[the remark this answers is not in the document]
Comment (Davknapp) [reply test/t8_gtest_custom_assertion.hxx:99]: Good idea

**Review by lukasdreyer — APPROVED** [review]

**Pull request merged**